### PR TITLE
Select the correct internal method for foldArguments

### PR DESCRIPTION
**Pull request by ChengJin01**
The test is updated to set the expected internal method
based on the java version.

Fixes: #12980

Signed-off-by: Cheng Jin <jincheng@ca.ibm.com>

### CODE REVIEW

**Comment by ChengJin01** [comment]
Verified with internal builds (Java 11, 12 and 16)

Reviewer: @tajila, @pshipton
FYI:  @fengxue-IS, @babsingh

File: test/functional/Jsr292/src/com/ibm/j9/jsr292/api/MethodHandleAPI_foldArguments.java [hunk 1]
```diff
@@ -30,11 +30,12 @@
 import static java.lang.invoke.MethodType.methodType;
 import java.lang.reflect.InvocationTargetException;
 import java.lang.reflect.Method;
+import org.openj9.test.util.VersionCheck;
 
 public class MethodHandleAPI_foldArguments {
 	final Lookup lookup = lookup();
 	final String MHClassName = "java.lang.invoke.MethodHandles";
-
+	final String internalFoldArgumentMethod = (VersionCheck.major() >= 12) ? "foldArgumentsWithCombiner" : "foldArguments";
```
Comment (tajila) [inline test/functional/Jsr292/src/com/ibm/j9/jsr292/api/MethodHandleAPI_foldArguments.java:38]: I think you can just to `VersionCheck.major() > 11`

Comment (tajila) [reply test/functional/Jsr292/src/com/ibm/j9/jsr292/api/MethodHandleAPI_foldArguments.java:38]: jdk12 is obsolete

Comment (ChengJin01) [reply test/functional/Jsr292/src/com/ibm/j9/jsr292/api/MethodHandleAPI_foldArguments.java]: Agreed and updated as suggested.

**Review by tajila — APPROVED** [review]

**Comment by tajila** [comment]
Jenkins test sanity plinux jdk11,jdk16,jdk8

**Comment by pshipton** [comment]
Note the tests are extended tests, I don't believe sanity exercises this code. I can run grinders on the builds once they are completed, I'll cancel the sanity tests.

**Comment by pshipton** [comment]
Actually we don't need the PR builds, nothing in the JVM changed.

**Comment by pshipton** [comment]
Grinders on jsr292_MethodHandleAPI_Test_JitCount1_0,jsr292_MethodHandleAPI_Test_0 passed. These tests don't run on jdk8.
jdk11 https://openj9-jenkins.osuosl.org/view/Test/job/Grinder/76
jdk16 https://openj9-jenkins.osuosl.org/view/Test/job/Grinder/78

**Review by pshipton — APPROVED** [review]

**Pull request merged**